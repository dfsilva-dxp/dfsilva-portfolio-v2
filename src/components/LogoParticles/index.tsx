import useWindowSize from "@rooks/use-window-size";
import ParticleImage, {
  ParticleOptions,
  Vector,
  forces,
  ParticleForce
} from "react-particle-image";

import logo from "../../assets/logo.svg";

const particleOptions: ParticleOptions = {
  filter: ({ x, y, image }) => {
    const pixel = image.get(x, y);
    return pixel.b > 5;
  },
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  color: ({ x, y, image }) => "#020015",
  radius: () => Math.random() * 0.5 + 0.5,
  mass: () => 10,
  friction: () => 0.35,
  initialPosition: ({ canvasDimensions }) => {
    return new Vector(canvasDimensions.width / 2, canvasDimensions.height / 2);
  }
};

const motionForce = (x: number, y: number): ParticleForce => {
  return forces.disturbance(x, y, 5);
};

const LogoParticles = () => {
  const { innerWidth, innerHeight } = useWindowSize();

  return (
    <ParticleImage
      src={logo}
      width={Number(innerWidth)}
      height={Number(innerHeight)}
      scale={1}
      entropy={0.15}
      maxParticles={30000}
      particleOptions={particleOptions}
      mouseMoveForce={motionForce}
      touchMoveForce={motionForce}
      backgroundColor="#CB7F65"
    />
  );
};

export default LogoParticles;
