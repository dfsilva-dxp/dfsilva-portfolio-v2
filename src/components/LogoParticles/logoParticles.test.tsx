import { screen } from "@testing-library/react";
import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import LogoParticles from ".";

describe("<LogoParticles/>", () => {
  it("should render currectly", () => {
    const { container } = renderWithTheme(<LogoParticles />);

    expect(screen.getByLabelText("Logo Particles")).toHaveStyle({
      "background-color": "transparent"
    });

    expect(container).toMatchSnapshot();
  });
});
