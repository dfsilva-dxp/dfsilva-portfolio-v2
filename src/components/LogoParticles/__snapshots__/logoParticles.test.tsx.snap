// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoParticles/> should render currectly 1`] = `
<div>
  <canvas
    aria-label="Logo Particles"
    height="648"
    style="background-color: transparent;"
    width="1024"
  />
</div>
`;
